gio/src/file.rs

```diff
@@ -1111,6 +1113,151 @@ pub trait FileExtManual: IsA<File> + Sized {
 
         (fut, Box::pin(receiver))
     }
+
+    #[doc(alias = "g_file_set_attribute")]
+    fn set_attribute<'a>(
+        &self,
+        attribute: &str,
+        value: impl Into<FileAttributeValue<'a>>,
+        flags: FileQueryInfoFlags,
+        cancellable: Option<&impl IsA<Cancellable>>,
+    ) -> Result<(), glib::Error> {
+        unsafe {
+            let mut error = std::ptr::null_mut();
+            let value: FileAttributeValue<'a> = value.into();
+            let is_ok = ffi::g_file_set_attribute(
+                self.as_ref().to_glib_none().0,
+                attribute.to_glib_none().0,
+                value.type_().into_glib(),
+                value.as_ptr(),
+                flags.into_glib(),
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                &mut error,
+            );
+            debug_assert_eq!(is_ok == glib::ffi::GFALSE, !error.is_null());
+            if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
 }
 
 impl<O: IsA<File>> FileExtManual for O {}
+
+#[derive(Debug)]
+pub struct FileAttributeValue<'a>(FileAttributeValueInner<'a>);
+
+impl From<&str> for FileAttributeValue<'_> {
+    fn from(value: &str) -> Self {
+        Self(FileAttributeValueInner::String(
+            ToGlibPtr::<*mut libc::c_char>::to_glib_none(value).1,
```

Comment:
This creates a dangling pointer. `ToGlibPtr` creates a new temporary NUL-terminated string that is then freed when the `Stash` is dropped

Reply:
You could e.g. internally store a `glib::GString` instead and on `as_ptr()` use that directly

Reply:
Oh you're actually storing the `Stash` here. `.1` and not `.0`. Nevermind then :)